### PR TITLE
Implemented politeness policy for the crawler

### PR DESCRIPTION
**Pull request by mustiikhalil**

### CODE REVIEW

File: crawler.rb [hunk 1]
```diff
@@ -43,9 +43,12 @@ def start_crawling()
         # save to Datastore
         add_to_datastore(crawled_page.url.to_s, crawled_page.main_divs.to_s)
 
+        # Crawler Politness
```
Comment (heshanka) [inline crawler.rb]: please address the typo

**Comment by evis-e** [comment]
I was thinking that we can also do this differently: since we are skipping a lot of URLs, we are not sending too many requests to the servers anyways. Could we consider crawling without sleeping, and once we get an exception, catch it, sleep 30 seconds, and keep crawling again?

**Comment by mustiikhalil** [comment]
@evis-e  well that's another way of doing it for sure. but we still need to get a small request break out of courtesy to the companies. We can make it 5 sec, if an expectation is raised then handle it and sleep 30 sec

**Review by heshanka — APPROVED** [review]

**Comment by evis-e** [comment]
If we want to send a request every 30 seconds, we could calculate how much time passed since the last request, and wait the remaining amount?

**Review by evis-e — APPROVED** [review]

**Pull request merged**